src/app.ts

```diff
@@ -390,7 +452,16 @@ export default class ChessGame {
 					.filter(item => item.file === dropSquare.file && item.rank === dropSquare.rank).shift();
 				if (destSquare) {
 					// Move the piece.
-					this.game.move(move.src, destSquare);
+					if (move.src.piece.type === 'pawn' && (destSquare.rank === 8 || destSquare.rank === 1)) {
+						//Auto promote pawn to queen
+						this.game.move(move.src,destSquare, "Q");
+						
+						const newActor = this.promoteChessPiece(userId, actor, destSquare);
+						await Promise.resolve(newActor);

```

Comment:
Why is the call to Promise.resolve necessary? Would this work:
```ts
actor = this.promoteChessPiece(userId, actor, destSquare);
```

Reply:
Yes that does work. I'm pushing the change now